### PR TITLE
fix listServiceConfigurationUpdateRequest method

### LINKED ISSUE #1964: fix listServiceConfigurationUpdateRequest method
listServiceConfigurationUpdateRequest method response is not very useful. 

- [ ] rename to getAllServiceConfigurationUpdateRequests
- [ ] URI - /services/config/requests
- [ ] make serviceId in the request query parameter as mandatory
- [ ] Response must contain -> 
[
  orderId: xxxxxx
  changeRequests: [
  {
    "changeId": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
    "resourceName": "string",
    "configManager": "string",
    "resultMessage": "string",
    "properties": {
      "additionalProp1": {}
    },
    "status": "pending"
  }
	]
]

### PR DESCRIPTION
**Pull request by WangLiNaruto**
Fixed #1964 
image
image

### CODE REVIEW

**Comment by Alice1319** [comment]
LGTM

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/serviceconfiguration/ServiceConfigurationUpdateRequestOrderDetails.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ */
+@Data
+public class ServiceConfigurationUpdateRequestOrderDetails {
+
+    @NotNull
+    @Schema(description = "The id of the order.")
+    private UUID orderId;
+
+    @NotNull
+    @Schema(description = "Collection of service configuration change request.")
```
Comment (swaroopar) [inline modules/models/src/main/java/org/eclipse/xpanse/modules/models/serviceconfiguration/ServiceConfigurationUpdateRequestOrderDetails.java:25]: change description - Collection of service configuration change requests generated for the specific change order.

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/serviceconfiguration/ServiceConfigurationChangeRequestDetails.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import org.eclipse.xpanse.modules.models.serviceconfiguration.enums.ServiceConfigurationStatus;
+
+/**
+ * service configuration change request details.
+ */
+@Data
+public class ServiceConfigurationChangeRequestDetails {
+
+    @NotNull
+    @Schema(description = "id of service configuration update request.")
```
Comment (swaroopar) [inline modules/models/src/main/java/org/eclipse/xpanse/modules/models/serviceconfiguration/ServiceConfigurationChangeRequestDetails.java:22]: ID of the change request created as part of the change order.

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/serviceconfiguration/ServiceConfigurationChangeRequestDetails.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    private UUID changeId;
+
+    @Schema(description = "name of the resource of deployed service.")
```
Comment (swaroopar) [inline modules/models/src/main/java/org/eclipse/xpanse/modules/models/serviceconfiguration/ServiceConfigurationChangeRequestDetails.java:25]: name of the resource on which the change request is executed. Null means any one of the resources that is part of the service and is of type configManager can execute it and until now none of the resource have picked up this request.

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/serviceconfiguration/ServiceConfigurationChangeRequestDetails.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    private String resourceName;
+
+    @NotNull
+    @Schema(description = "the name available in the deployer script.")
```
Comment (swaroopar) [inline modules/models/src/main/java/org/eclipse/xpanse/modules/models/serviceconfiguration/ServiceConfigurationChangeRequestDetails.java:29]: type of the resource in the service that must execute the change request.

File: modules/models/src/main/java/org/eclipse/xpanse/modules/models/serviceconfiguration/ServiceConfigurationChangeRequestDetails.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    private String configManager;
+
+    @Schema(description = "message of change service configuration update request.")
+    private String resultMessage;
+
+    @NotNull
+    @Schema(description = "properties of service configuration.")
```
Comment (swaroopar) [inline modules/models/src/main/java/org/eclipse/xpanse/modules/models/serviceconfiguration/ServiceConfigurationChangeRequestDetails.java:36]: service configuration requested in the change request.

**Review by swaroopar — CHANGES_REQUESTED** [review]

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**